core/services/ocr2/plugins/mercury/plugin.go

```diff
@@ -46,23 +50,47 @@ func NewServices(
 		return nil, err
 	}
 	lggr = lggr.Named("MercuryPlugin").With("jobID", jb.ID, "jobName", jb.Name.ValueOrZero())
-	ds := mercury.NewDataSource(
-		pipelineRunner,
-		jb,
-		*jb.PipelineSpec,
-		lggr,
-		runResults,
-		chEnhancedTelem,
-		chainHeadTracker,
-		ocr2Provider.ContractTransmitter(),
-		pluginConfig.InitialBlockNumber.Ptr(),
-	)
-	argsNoPlugin.MercuryPluginFactory = relaymercury.NewFactory(
-		ds,
-		lggr,
-		ocr2Provider.OnchainConfigCodec(),
-		ocr2Provider.ReportCodec(),
-	)
+
+	switch ocr2Provider.ReportSchemaVersion() {
+	case 2:
+		ds := mercuryv2.NewDataSource(
+			pipelineRunner,
+			jb,
+			*jb.PipelineSpec,
+			lggr,
+			runResults,
+			chEnhancedTelem,
+			chainHeadTracker,
+			ocr2Provider.ContractTransmitter(),
+			pluginConfig.InitialBlockNumber.Ptr(),
+		)
+		argsNoPlugin.MercuryPluginFactory = relaymercuryv2.NewFactory(
+			ds,
+			lggr,
+			ocr2Provider.OnchainConfigCodec(),
+			ocr2Provider.ReportCodecV2(),
+		)
+	default:
+		// defaulting to v1 for backwards compatibility
```

Comment:
I would suggest instead starting the version number at `0`. So you have versions 0, 1, 2 etc

This works out nicely because if `reportSchemaVersion` is unset it will be the zero value, so it will automatically be version 0 anyway. Then you don't need this `default:` clause (or you can panic for an unsupported version).